File: php/testcode/securecookie_009.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_cookie_samesite_none_insecure
function securecookie009(BenchmarkRequest $req): BenchmarkResponse {
    $token = bin2hex(random_bytes(32));
    setcookie('session', $token, [
        'expires' => time() + 3600,
        'path' => '/',
        'samesite' => 'None',
        'secure' => false, // vuln-code-snippet vuln-line php_cookie_samesite_none_insecure
        'httponly' => true,
    ]);
    return BenchmarkResponse::ok('Session started');
}
// vuln-code-snippet end php_cookie_samesite_none_insecure

File: rust/testcode/securecookie_009.rs
//! CWE-614: Session cookie with Path=/ but no domain restriction or Secure flag.

// vuln-code-snippet start testcodeSecurecookie009
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let token = req.param("token");

    let cookie = build_session_cookie(&token); // vuln-code-snippet target-line testcodeSecurecookie009

    super::shared::BenchmarkResponse::ok(&format!("Set-Cookie: {}", cookie))
}

fn build_session_cookie(value: &str) -> String {
    format!("session={}; Path=/; HttpOnly", value)
}
// vuln-code-snippet end testcodeSecurecookie009
